api/machinemanager/machinemanager.go

```diff
@@ -133,5 +133,24 @@ func (client *Client) UpdateMachineSeries(machineName, series string, force bool
 	if err != nil {
 		return errors.Trace(err)
 	}
+
+	return results.OneError()
+}
+
+// SeriesUpgradePrepare notifies the controller that a series upgrade is taking
+// place for a given machine and as such the machine is guarded against
+// operations that would impede, fail, or interfere with the upgrade process.
+func (client *Client) UpgradeSeriesPrepare(machineName string) error {
+	if client.BestAPIVersion() < 5 {
+		return errors.NotSupportedf("upgrade-series prepare")
+	}
+	args := params.UpdateSeriesArg{
```

Comment:
Thank you @hmlanigan, we could of course do this but I am just following the best practice of not doing more than is needed since the client doesn't not yet require a bulk call, so I did not make one. We could of course turn it into a bulk call quite easily if desired. Let me know how you feel.

Reply:
In code reviews I'm usually asked to make the apiserver calls work in bulk.  The calls this supersedes are also in bulk already, so it looks like there is already a precedent in this case.

Reply:
Will do. Lets hold off on this though because this PR is getting huge. I will issue a follow-up PR that address concerns like this specifically.